test/liquidity-protection/LiquidityProtection.js

```diff
@@ -3802,6 +3426,536 @@ describe('LiquidityProtection', () => {
+                            prevStates[ethToken.address].providerBaseBalance =
+                                prevStates[ethToken.address].providerBaseBalance.sub(gasCost);
+
+                            for (const baseToken of [xxxToken, yyyToken, ethToken]) {
+                                const basePositions = getPositions(positions, baseToken.poolToken, baseToken);
+                                const networkPositions = getPositions(positions, baseToken.poolToken, networkToken);
+                                const baseReserveAmount = basePositions.reduce(
+                                    (a, b) => a.add(b.reserveAmount),
+                                    BigNumber.from(0)
+                                );
+                                const networkReserveAmount = networkPositions.reduce(
+                                    (a, b) => a.add(b.reserveAmount),
+                                    BigNumber.from(0)
+                                );
+                                const basePoolAmount = basePositions.reduce(
+                                    (a, b) => a.add(b.poolAmount),
+                                    BigNumber.from(0)
+                                );
+                                const networkPoolAmount = networkPositions.reduce(
+                                    (a, b) => a.add(b.poolAmount),
+                                    BigNumber.from(0)
+                                );
+                                const totalPoolAmount = basePoolAmount.add(networkPoolAmount);
+                                const deltaPoolAmount = basePoolAmount.sub(networkPoolAmount);
+
+                                const prevState = prevStates[baseToken.address];
+                                const currState = await readState(baseToken, providers[i]);
+
+                                expect(currState.providerGovBalance).to.equal(prevState.providerGovBalance);
+                                expect(currState.providerBaseBalance).to.equal(prevState.providerBaseBalance);
+                                expect(currState.providerNetworkBalance).to.equal(prevState.providerNetworkBalance);
+                                expect(currState.providerPoolBalance).to.equal(prevState.providerPoolBalance);
+                                expect(currState.vaultGovBalance).to.equal(prevState.vaultGovBalance);
+                                expect(currState.vaultPoolBalance).to.equal(prevState.vaultPoolBalance);
+                                expect(currState.walletGovBalance).to.equal(prevState.walletGovBalance);
+                                expect(currState.walletBaseBalance).to.equal(prevState.walletBaseBalance);
+                                expect(currState.walletNetworkBalance).to.equal(prevState.walletNetworkBalance);
+
+                                expect(currState.vaultNetworkBalance).to.equal(networkReserveAmount.mul(3)); // TODO: investigate
+
```

Comment:
TODO: investigate?

Reply:
Continuing our call, you don't really need to check v3 balances in these tests

Reply:
Contract `TestBancorNetwork` mocks function `migrateLiquidity`, which merely transfers the tokens to some preconfigured "vault" address.
These extra checks are for ensuring that the function was called with correct amounts.
In a sense, they are partially redundant (because the mock function could do anything that we'd put into it).
However, they still have some added value in verifying that the real function is being called (internally from function `migratePositions`) with the correct input.
For this added value, I think that we should keep these extra checks.

Reply:
if we could check that the migrateLiquidity function is called with the proper input, that should be a sufficient, as we're testing the liquidity protection migration logic here, not the v3 network contract

Reply:
As I've explained above. this is exactly what we're doing here (checking that the `migrateLiquidity` function is called with the proper input).